R/RunAnalyses.R

```diff
@@ -892,100 +920,118 @@ applyTrimMatchStratify <- function(ps, arguments) {
 }
 
 doTrimMatchStratify <- function(params) {
-  ps <- getPs(params$psFile)
-  ParallelLogger::logDebug(sprintf("Performing matching etc., using %s",
-                                   params$psFile))
-  ps <- applyTrimMatchStratify(ps, params$args)
-  saveRDS(ps, params$strataFile)
+  tryCatch({
+    ps <- getPs(params$psFile)
+    ParallelLogger::logDebug(sprintf("Performing matching etc., using %s",
+                                     params$psFile))
+
+
+    ps <- applyTrimMatchStratify(ps, params$args)
+    saveRDS(ps, params$strataFile)
+  }, error = function(err) {
+    ParallelLogger::logError(sprintf("Unable to save %s - %s", params$strataFile, err))
+  })
   return(NULL)
 }
 
 doPrefilterCovariates <- function(params) {
-  cohortMethodData <- loadCohortMethodData(params$cohortMethodDataFile)
-  covariates <- cohortMethodData$covariates
-  if (nrow_temp(covariates) > 0) {
-    if (params$args$useCovariates) {
-      covariatesToInclude <- params$args$includeCovariateIds
-      covariatesToExclude <- params$args$excludeCovariateIds
-    } else {
-      covariatesToInclude <- c()
-      covariatesToExclude <- c()
-    }
-    covariatesToInclude <- unique(c(covariatesToInclude, params$args$interactionCovariateIds))
-    if (length(covariatesToInclude) != 0) {
-      covariates <- covariates %>%
-        filter(.data$covariateId %in% covariatesToInclude)
-    }
-    if (length(covariatesToExclude) != 0) {
-      covariates <- covariates %>%
-        filter(!.data$covariateId %in% covariatesToExclude)
+  tryCatch({
```

Comment:
Double yikes! What is the point in wrapping such a large block of code in a try catch?